raft/raft.go

```diff
@@ -281,6 +281,9 @@ type raft struct {
 	tick func()
 	step stepFunc
 
+	// voterCount caches the total voter which can vote.
+	voterCount int
```

Comment:
i still want to try to separate the learners out of the prs. so we can keep the old logic without adding this "cache". i feel the code can be cleaner overall too. 

can you give it a try at least?